### PR TITLE
Add Dotty error messages post

### PR DESCRIPTION
**Pull request by felixmulder**
Hey there!

We'd like to add this post on dotty error messages to the blog.

Cheers,
Felix

### CODE REVIEW

**Comment by felixmulder** [comment]
ping: @SethTisue

File: blog/_posts/2016-10-14-dotty-errors.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+Let's dive right into some examples, let's say you have this code:
+
+    try {
+      1 + 2
+    }
+
+It doesn't really make sense to put this in a `try`-block for two reasons:
+
+1. It doesn't throw an exception
```
Comment (sjrd) [inline blog/_posts/2016-10-14-dotty-errors.md]: This part is irrelevant to the present discussion. I suggest you replace `1 + 2` by `foo()`, and let the reader figure out that `foo()` might throw an exception.

File: blog/_posts/2016-10-14-dotty-errors.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+<pre>
+test.scala:2: warning: A try without a catch or finally is equivalent to putting its body in a block; no exceptions are handled.
+     try {
+       ^
+       one warning found
+</pre>
+
+This is helpful, but it has a couple of drawbacks:
+
+1. If we have a bazillion errors, it will be hard to read
```
Comment (sjrd) [inline blog/_posts/2016-10-14-dotty-errors.md]: Is that really relevant, here?

File: blog/_posts/2016-10-14-dotty-errors.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+compiler as you'll need to create semantic objects that contain the relevant
+information for the error or warning.
+
+So - this is what you do:
+
+1. Go to the [Error messages issue](https://github.com/lampepfl/dotty/issues/1589)
+2. Read the howto on error messages
+3. Choose an error message you want to help with and post a comment saying
+   which one(s)
+4. Get hacking!
```
Comment (sjrd) [inline blog/_posts/2016-10-14-dotty-errors.md]: > 5 - Submit a PR

?

File: blog/_posts/2016-10-14-dotty-errors.md (hunk 2 above)
Comment (felixmulder) [inline blog/_posts/2016-10-14-dotty-errors.md]: Yes - the new errors have a sane division between errors so you can see where one ends and another begins.

File: blog/_posts/2016-10-14-dotty-errors.md (hunk 1 above)
Comment (felixmulder) [inline blog/_posts/2016-10-14-dotty-errors.md]: Alright!

**Comment by heathermiller** [comment]
👏 Can I merge this now?

**Comment by heathermiller** [comment]
Well OK, I'm doing it :)

**Comment by SethTisue** [comment]
> ping: @SethTisue

what's the equivalent of "the squeaky wheel gets the grease", but for the person who oils the wheels? "the squeaky wheel only squeaks when the guy with the oil can walks by"?

**Pull request merged**